### PR TITLE
Pet Features

### PR DESCRIPTION
**Pull request by javier-aguilar**
- Pets with approved applications cannot be deleted
- Deleting a pet removes it from favorites
- Flash Message for Pet Create and Update

### CODE REVIEW

File: app/controllers/pets_controller.rb [hunk 1]
```diff
@@ -29,8 +29,12 @@ def edit
   def update
     pet = Pet.find(params[:id])
     pet.update(pet_params)
-    pet.save
-    redirect_to "/pets/#{pet.id}"
+    if pet.save
+      redirect_to "/pets/#{pet.id}"
```
Comment (tylerpporter) [inline app/controllers/pets_controller.rb]: These first few lines can be condensed into ` if Pet.update(params[:id], pet_params).save`

File: app/controllers/pets_controller.rb [hunk 2]
```diff
@@ -67,4 +77,18 @@ def pet_params
                   :sex)
   end
 
+  def delete_pet_apps(pet)
+    pet.apps.each { | app | pet.apps.delete(app) } if pet.apps.present?
+  end
```
Comment (tylerpporter) [inline app/controllers/pets_controller.rb]: I think this logic should probably be executed on the model level. So in the Pet model, have the method 
```ruby 
def delete_apps 
  apps.destroy_all if apps.present?
end 
```
Then in the `destroy` method, we can just call `pet.delete_apps`

Comment (tylerpporter) [reply app/controllers/pets_controller.rb]: But also, not entirely sure we need any of this logic.... won't just including a `dependent: :destroy` on the `has_many :pet_apps` relationship take care of destroying associated apps?

Comment (javier-aguilar) [reply app/controllers/pets_controller.rb]: Ah yes. I keep forgetting about that 🙃

File: app/controllers/pets_controller.rb (hunk 1 above)
Comment (javier-aguilar) [reply app/controllers/pets_controller.rb]: Yep. I did this for the other update methods as well

**Comment by tylerpporter** [comment]
Awesome, this looks great. Thanks for addressing the comments I left!

**Pull request merged**